cibuildwheel/windows.py

```diff
@@ -154,6 +154,13 @@ def setup_setuptools_cross_compile(
     # set the platform name
     map_plat = {"32": "win32", "64": "win-amd64", "ARM64": "win-arm64"}
     plat_name = map_plat[python_configuration.arch]
+
+    # Set environment variable so that setuptools._distutils.get_platform()
+    # identifies the target, not the host
+    if not env.get("VSCMD_ARG_TGT_ARCH"):
+        vscmd_arg_tgt_arch = {"32": "x86", "64": "x64", "ARM64": "arm64"}
+        env["VSCMD_ARG_TGT_ARCH"] = vscmd_arg_tgt_arch[python_configuration.arch]
```

Comment:
`env.setdefault` would be more ideomatic I think, that's what we use in macos.py

Reply:
If this is already set by `vcvarsall.bat`, then setdefault wouldn't change it, and it would be wrong.

Reply:
The behaviour in the diff is to only set if not set already. I was only suggesting a style change really. But...

Who is calling `vcvarsall.bat`? I don't think cibuildwheel calls it. Do some users call that before calling cibuildwheel?

Reply:
Ahh, didn't look up... 2 lines. Yes, set default is better. Though is this what we want? Would erroring out, for example, make more sense?

Users may set up MSVC before calling cibuildwheel. [here are 366 examples](https://github.com/search?type=code&q=path%3A.github+msvc-dev-cmd+cibuildwheel).

Reply:
[`vcvarsall.bat`](https://learn.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=msvc-170#use-the-developer-tools-in-an-existing-command-window) and [`msvc-dev-cmd`](https://github.com/ilammy/msvc-dev-cmd?tab=readme-ov-file#inputs) do allow themselves to be configured for different targets, including cross-compilation. So asking cibuildwheel to cross-compile after calling one of these tools for a different target feels like a misconfiguration IMO.

Yes, perhaps an error here would be better. If VSCMD_ARG_TGT_ARCH is set to something other than what we were gonna set, we should error out IMO. The user can then decide what they want to do - either stop using vcvarsall/msvc-dev-cmd or call it again with the right configuration before cross-compiling.

Reply:
Agree.

Reply:
> Yes, perhaps an error here would be better. If VSCMD_ARG_TGT_ARCH is set to something other than what we were gonna set, we should error out IMO. 

Yep, if a user has run vcvarsall already, _that's_ the cross-compiling configuration done. If they want cibuildwheel to configure it, then they shouldn't do it themselves. Otherwise you really need to clear all (most of) the environment variables that were set so that the build backend knows to reconfigure.